doc/scripts/filter

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/perl
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2013 - 2019 by the deal.II authors
+##
+## The original file is from the deal.II library. It has been modified for
+## Ginkgo.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE.md at
+## the top level directory of deal.II.
+##
+## ---------------------------------------------------------------------
+
+# read all lines of input. within the loop, first do some
+# easy substitutions that are confined to a single line; later
+# do some multi-line substitutions as well
+
+while (<>)
+{
+    ########################################################
+    # Step 1: substitutions within a single line
+    ########################################################
+
+
+    # make sure we can just write $...$ for formulas.
+    s/\$/\@f\$/g;
+
+    # however, undo that change if the dollar sign was escaped with a backslash
+    s/\\\@f\$/\$/g;
+
+    # Replace all occurrences of something like example-xx by
+    #    @ref example_xx "example-xx"
+    # so that doxygen properly cross references them. Before we had
+    # this rule, we actually had to write this sequence out in our
+    # documentation. Unfortunately, as a consequence, there are vestiges
+    # of this style, so we can't substitute things that look like
+    # "example-xx". We therefore not substitute if example-xx is preceded or
+    # followed by quotation marks, or if the text is explicitly
+    # preceded by a backslash for escaping.
+    #
+    # There are other exceptions:
+    # - the scripts in doc/doxygen/tutorial produce files that have
+    #   table of contents entries. We don't want these cross-linked
+    #   to itself.
+    # - things like custom-stopping-criterion.solution.png that typically appear in
+    #   @image commands.
+    # - things in headings
+    s/(?<![\"\\\/])example-(\w+)(?!\")/\@ref example_\1 \"example-\1\"/gi
+        if !m/(\@page|\<img|\@image|<h\d>)/i;
+
+    # If example-xx was explicitly escaped with a backslash, remove the
+    # latter
+    s/\\(example-\w+)/\1/g;
```

Comment:
This maybe needs to be updated as this uses the old `example-<number>` notation. In the whole file.

Or maybe you do not call this anyway, in which case feel free to remove it (and any other script you do not use if any). It looks like you do not use this as this is also supposed to change the `\f$` to `$`.